### PR TITLE
X-ray settings toggle 

### PR DESCRIPTION
**Pull request by kdoh**
Adds in a setting in the `General` settings in the admin section to disable x-rays for an instance. Does so by hiding UI triggers that direct users to routes that use x-ray resources. This setting defaults to "enabled" and the primary motivation here is to help out Metabase users who are dealing with data sources where allowing users to run x-rays on them incurs performance or monetary costs by giving them the ability to effectively disable the feature.

TODOs

- [x] Hide x-rays in click actions

### CODE REVIEW

File: frontend/src/metabase/admin/settings/selectors.js [hunk 1]
```diff
@@ -511,3 +516,9 @@ export const getActiveSection = createSelector(
     }
   },
 );
+
+export const getXraysEnabled = createSelector(getSettingValues, settings => {
```
Comment (tlrobinson) [inline frontend/src/metabase/admin/settings/selectors.js]: I'd put this in `frontend/src/metabase/selectors/settings.js` with `getIsPublicSharingEnabled` etc. It also needs to be added to `public-settings` in `src/metabase/public_settings.clj`

File: frontend/src/metabase/admin/settings/selectors.js [hunk 2]
```diff
@@ -511,3 +516,9 @@ export const getActiveSection = createSelector(
...
+  // apparently "null" is true for boolean settings so we need to check for
```
Comment (tlrobinson) [inline frontend/src/metabase/admin/settings/selectors.js]: Wait, what?

File: frontend/src/metabase/qb/components/actions/XRayCard.jsx [hunk 3]
```diff
@@ -8,7 +8,10 @@ import type {
 import { t } from "c-3po";
 import { utf8_to_b64url } from "metabase/lib/card";
 
-export default ({ question }: ClickActionProps): ClickAction[] => {
+export default ({ question, settings }: ClickActionProps): ClickAction[] => {
```
Comment (tlrobinson) [inline frontend/src/metabase/qb/components/actions/XRayCard.jsx]: I don't think `settings` is included in `ClickActionProps`?

Probably easiest to just do `MetabaseSettings.get("enable_xrays")`

**Comment by mazameli** [comment]
I think we should hide `Start Here` if there are no x-rays or pinned dashboards:

screen shot 2018-10-17 at 11 45 25 am

**Comment by mazameli** [comment]
Verified that turning the x-rays setting off does indeed hide all of the x-ray triggers that I could remember:

- on the homepage when you don’t have any pinned dashboards
- on table links in the Data Browse screen
- in the bottom-right action button in the query builder
- in the drill-through menu when clicking on a chart
- in the data reference sidebar

The main open question that I recall is still whether or not we also turn this feature off on the backend so that you can't hit an x-ray URL directly. Did we land on a decision WRT that?

**Comment by mazameli** [comment]
Ah, I knew I forgot something: there are also x-ray links in the data reference for Segments, Metrics, and Fields, and these are not being hidden currently.

**Comment by salsakran** [comment]
imo nixing it on the frontend is sufficient

**Review by tlrobinson — COMMENTED** [review]
Looks fine to me, though I wonder if we should just use `MetabaseSettings.get("enable_xrays")` for things like this where it's unlikely or impossible the state to change while you're displaying those components (i.e. that setting isn't going to change while you're viewing anything except the settings panel). It just seems like you had to `connect` a lot of components that weren't otherwise connected.

If we do that I'd probably add a `feature` function to settings so it would be something like `MetabaseSettings.feature("xrays")`.

**Comment by mazameli** [comment]
Looks like all the data reference x-ray links, including for segments, metrics, and fields are now correctly hidden. Also looks like the `start here` heading is correctly hidden. So I say ship it 👍

**Pull request merged**